source/distributed/cell_weights.cc

```diff
@@ -145,18 +145,23 @@ namespace parallel
             std::set<unsigned int> fe_indices_children;
             for (unsigned int child_index = 0; child_index < cell->n_children();
                  ++child_index)
-              fe_indices_children.insert(
-                cell->child(child_index)->future_fe_index());
+              {
+                const auto child = cell->child(child_index);
+                Assert(child->active() && child->coarsen_flag_set(),
+                       typename dealii::Triangulation<
+                         dim>::ExcObscureCoarseningFlags());
+
+                fe_indices_children.insert(child->future_fe_index());
+              }
+            Assert(!fe_indices_children.empty(), ExcInternalError());
 
             fe_index =
               dof_handler->get_fe_collection().find_dominated_fe_extended(
                 fe_indices_children, /*codim=*/0);
 
             Assert(fe_index != numbers::invalid_unsigned_int,
-                   ExcMessage(
-                     "No FiniteElement has been found in your FECollection "
-                     "that is dominated by all children of a cell you are "
-                     "trying to coarsen!"));
+                   typename dealii::hp::FECollection<
+                     dim>::ExcNoDominatedFiniteElementAmongstChildren());
```

Comment:
Do you think placing this particular one in `hp::FECollection<dim>` is reasonable?